contoso_manufacturing/developer/webapp-decode/yolov8.py

```diff
@@ -22,30 +24,57 @@ def __init__(self, rtsp_url, class_names, input_shape, color_palette, confidence
         self.verbose=verbose
         self.frame_number =0
         self.skip_rate=skip_rate
-
-        self.cap = cv2.VideoCapture(rtsp_url)
         self.grpc_client = make_grpc_client(ovms_url)
+        self.stopped = False
+        self.lock = threading.Lock()
+        self.frame_queue = queue.Queue(maxsize=50)
+        self.capture_thread = threading.Thread(target=self.capture_frames)
+        self.capture_thread.start()
+
+    def capture_frames(self):
+        cap = cv2.VideoCapture(self.rtsp_url)
+        while not self.stopped:
+            ret, frame = cap.read()
+            if not ret:
+                print("Failed to grab frame")
+                break
+            if not self.frame_queue.full():
```

Comment:
It would be good to know if the queue is overflows.. could we log an informational message here if that happens?
I think this could probably be a verbose log (self.log) rather than an error (print) since the code should handle this gracefully by just dropping the frame and continuing.